apps/verticalhome/js/more_apps_navigation.js

```diff
@@ -2,84 +2,103 @@
 (function (exports) {
 
   var MoreAppsNavigation = {
-    controls: null,
+    items: null,
     selectedElemIndex: 0,
-    selectElement: null,
 
+    /**
+     * Initialize list of navigation in MoreAppsNavigation when all items are
+     * loaded
+     */
     init: function () {
-      this.controls = app.grid.getItems();
-      this.reset();
+      this.items = app.grid.getItems();
     },
 
+    /**
+     * Called when 'More Apps' screen has been opened. This function reset
+     * selected element on first element and add listener on keydown event
+     */
     reset: function () {
-      if (this.selectElement) {
-        this.selectElement.classList.remove('selected');
+      document.getElementsByTagName('gaia-grid-rs')[0].scrollTo(0, 0);
+      if (this.items[this.selectedElemIndex].element) {
+        this.items[this.selectedElemIndex].element.classList.remove('selected');
       }
       this.selectedElemIndex = 0;
-      this.selectElement = this.controls[this.selectedElemIndex].element;
-      this.selectElement.classList.add('selected');
-      this.clearNavigationEvents();
+      this.items[this.selectedElemIndex].element.classList.add('selected');
       window.addEventListener('keydown', this);
     },
 
-    clearNavigationEvents: function () {
+    /**
+     * Remove listener of keydown event, when "More Apps" screen is hide
```

Comment:
I think it is better to move "when "More Apps" screen is hidden" under something like "Note:" section in a comment.
So, overall description here might be:
"Removes listener of the keydown event.
Note: called when "More Apps" screen is closed"